<?xml version="1.0"?>
<launch>

  <node pkg="cob_3d_segmentation" type="qppf_node" name="segmentation_qppf" output="screen" >
  	<param name="filter" value="200" />
  	<param name="only_planes" value="true" />
  	
  	<!-- remap from="/camera/rgb/points" to="/cam3d/rgb/points"/ -->
  	<!-- remap from="/camera/rgb/camera_info" to="/cam3d/rgb/camera_info"/ -->
  	<!-- remap from="/camera/depth/camera_info" to="/cam3d/depth/camera_info"/ -->
  </node>

  <node pkg="cob_camera_alignment" type="camera_align" name="camera_align" output="screen" >
  	<rosparam file="$(find cob_camera_alignment)/config/walls.yaml" command="load" />
  	<param name="tf" value="/camera_depth_optical_frame" />
  	<param name="world" value="/base" />
  	<param name="threshold" value="0.45" />
  </node>

  <node pkg="cob_camera_alignment" type="start_calib.py" name="cob_calibration_controller" output="screen" >
  	<param name="number_of_frames" value="20" />
  	<param name="variance" value="0.001" />
  	<param name="max_delta_rad" value="0.2" />
  	
  	<param name="step_size" value="0.1" />
  	<param name="range_min" value="-3" />
  	<param name="range_max" value="10" />
  	
  	<param name="actor" value="cob_3d_mapping_demonstrator" />
  	<rosparam file="$(find cob_camera_alignment)/config/mapping_demonstrator_joints.yaml" command="load" />
  	<param name="joint_index" value="0" />
  </node>
  
  <include file="$(find cob_camera_alignment)/ros/launch/dashboard.launch"/>
  
</launch>

